pyfarm/master/user_interface/agents.py

```diff
@@ -73,11 +73,35 @@ def agents():
 
     agents_query = agents_query.order_by("%s %s" % (order_by, order_dir))
 
+    agents_count = agents_query.count()
+
+    per_page = int(request.args.get("per_page", 100))
+    page = int(request.args.get("page", 1))
+    filters["per_page"] = per_page
+    filters["page"] = page
+    num_pages = 1
+    all_pages = []
+    if per_page > 0:
+        agents_query = agents_query.offset((page - 1) * per_page).limit(per_page)
+        num_pages = int(agents_count / per_page)
+        if agents_count % per_page > 0:
+            num_pages = num_pages + 1
+        all_pages = range(0, num_pages)
```

Comment:
You probably want xrange here for Python < 3.x.  I'd suggest this at the top of the file

```python
try:
    range_ = xrange
except NameError:
    range_ = range
```